tests/dragonfly/replication_test.py

```diff
@@ -2336,6 +2336,30 @@ async def test_announce_ip_port(df_factory):
     assert port == "1337"
 
 
+@pytest.mark.asyncio
+async def test_replication_timeout_on_full_sync(df_factory: DflyInstanceFactory):
+    # setting replication_timeout to a very small value to force the replica to timeout
+    master = df_factory.create(replication_timeout=50, vmodule="replica=2,dflycmd=2")
+    replica = df_factory.create()
+
+    df_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    seeder = SeederV2(key_target=200_000)
+
+    seeder_task = asyncio.create_task(seeder.run(c_master, target_deviation=0.1))
+    await asyncio.sleep(2)  # wait for seeder running
+
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await seeder_task
+    await asyncio.sleep(1)
+
+    await assert_replica_reconnections(replica, 0)
+    df_factory.stop_all()
```

Comment:
No action needed just remember this is called when the fixture is destroyed. Clients are the ones that usually needed to be closed manually.